src/main.rs

```diff
@@ -382,6 +389,65 @@ impl App {
         }
     }
 
+    fn testcase(&self, args: &ArgMatches) -> Result<()> {
+        let handle = self.current_handle()?;
+        let clash = self.read_clash(&handle)?;
+
+        let mut ostyle = if args.get_flag("no-color") {
+            OutputStyle::plain()
+        } else {
+            OutputStyle::default()
+        };
+        if let Some(show_ws) = args.get_one::<bool>("show-whitespace") {
+            if *show_ws {
+                ostyle.input_whitespace = ostyle.input_whitespace.or(Some(ostyle.input));
+                ostyle.output_whitespace = ostyle.output_whitespace.or(Some(ostyle.output));
+            } else {
+                ostyle.input_whitespace = None;
+                ostyle.output_whitespace = None;
+            }
+        }
+
+        if let Some(values) = args.get_many::<u64>("TESTCASE_INDEX") {
+            let testcases_to_print: Vec<u64> = values.cloned().collect();
+
+            if testcases_to_print.is_empty() {
+                return Err(anyhow!("lul"))
+            }
+
+            // Return an error if any index is out of bounds
+            let num_tests = clash.testcases().len() as u64;
+            for idx in testcases_to_print.iter() {
+                if *idx > num_tests {
+                    return Err(anyhow!("Invalid index {idx}. The clash only has {num_tests} test cases."))
+                }
+            }
+
+            let show_in = args.get_flag("in");
```

Comment:
`only_in` and `only_out` might be better names for the current meaning of these variables.

Although I think a better idea would be 
```rust
let (show_in, show_out) = (!args.get_flag("out"), !args.get_flag("in"))
```
Which would make the variables mean what they say and would also simplify your logic down the line.